internal/mocks/mock_oidc_server.go

```diff
@@ -80,11 +83,24 @@ func (server mockOidcServer) start() {
 		}
 	})
 
+	server.httpServer = &http.Server{Addr: ":" + port, Handler: mockHandler}
```

Comment:
do we need to check if server.httpServer is nil?

Reply:
🤔 no, i'm creating that field here

Reply:
My question is more if it is possible to call start more than once?  If so, you will overwrite `server.httpServer` and stop will not shut down the previous http server.

Reply:
Gotcha. Refactored code so that if you call `start` twice by accident, the test fails:

```
=== RUN   TestBuildServerWithOIDCAuthenticationAlias
2024/05/10 10:35:35 failed to start mock OIDC server{error 26 0  listen tcp :63714: bind: address already in use}
```